### PR TITLE
Mainscreen

### PR DESCRIPTION
**Pull request by ct222ct**

### CODE REVIEW

File: .idea/androidTestResultsUserPreferences.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (lechld) [inline .idea/androidTestResultsUserPreferences.xml:2]: Do we need that file? Never seen it before. Should delete it if we don't require it.

File: .idea/deploymentTargetDropDown.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lechld) [inline .idea/deploymentTargetDropDown.xml:1]: same here

File: .idea/gradle.xml [hunk 3]
```diff
@@ -7,7 +7,6 @@
         <option name="testRunner" value="GRADLE" />
         <option name="distributionType" value="DEFAULT_WRAPPED" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
-        <option name="gradleJvm" value="11" />
```
Comment (lechld) [inline .idea/gradle.xml:10]: Okay, seems you have some other settings inside android studio than I. Let's revert that please.

File: .idea/misc.xml [hunk 4]
```diff
@@ -1,10 +1,17 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="ExternalStorageConfigurationManager" enabled="true" />
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" project-jdk-name="Android Studio default JDK" project-jdk-type="JavaSDK">
```
Comment (lechld) [inline .idea/misc.xml:4]: here too. I'll add that `.idea` folder to `.gitignore` so it won't show up next time

File: app/build.gradle [hunk 5]
```diff
@@ -137,8 +137,16 @@ dependencies {
     implementation "androidx.navigation:navigation-fragment-ktx:2.5.3"
     implementation "androidx.navigation:navigation-ui-ktx:2.5.3"
 
+
+    implementation "androidx.recyclerview:recyclerview-selection:1.1.0"
```
Comment (lechld) [inline app/build.gradle:141]: I'm not sure about that dependency changes. Maybe `recyclerView` makes sense

File: app/build.gradle [hunk 6]
```diff
@@ -137,8 +137,16 @@ dependencies {
...
+    implementation 'androidx.core:core-ktx:1.6.0'
```
Comment (lechld) [inline app/build.gradle:142]: already have that dependency, please remove

File: app/build.gradle [hunk 7]
```diff
@@ -137,8 +137,16 @@ dependencies {
...
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Comment (lechld) [inline app/build.gradle:143]: don't think we need that one

File: app/build.gradle [hunk 8]
```diff
@@ -137,8 +137,16 @@ dependencies {
...
+    implementation 'androidx.recyclerview:recyclerview:1.3.0'
```
Comment (lechld) [inline app/build.gradle:144]: same dependency twice

File: app/build.gradle [hunk 9]
```diff
@@ -137,8 +137,16 @@ dependencies {
...
+
     // https://developers.google.com/nearby
     implementation 'com.google.android.gms:play-services-nearby:18.5.0'
+    implementation 'androidx.test:core-ktx:1.4.0'
```
Comment (lechld) [inline app/build.gradle:148]: It's a testing library. `implementation` means we will add that library everywhere. For libs required for testing you should use `testImplementation`. That ensures that the dependency is not included in the application itself, but available when running tests.
But as you haven't written tests in this merge request this is an unnecessary chanmge. Please remove

File: app/src/main/res/values-night/styles.xml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+<resources>
+    <style name="Widget.Theme.Wizards.MyView" parent="">
```
Comment (lechld) [inline app/src/main/res/values-night/styles.xml:2]: when creating a new style, it should use the app theme as parent. Inside `Manifest` you can look up what we currently use as app theme.
But I think this style is not necessary for feature main screen/discover screen. Please remove.

File: app/src/main/res/values/attrs_my_view.xml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+<resources>
+    <declare-styleable name="MyView">
```
Comment (lechld) [inline app/src/main/res/values/attrs_my_view.xml:2]: please remove test code

File: app/src/main/res/values/colors.xml [hunk 12]
```diff
@@ -61,4 +61,8 @@
     <color name="md_theme_dark_surfaceTint">#D0BCFF</color>
     <color name="md_theme_dark_outlineVariant">#49454F</color>
     <color name="md_theme_dark_scrim">#000000</color>
+    <color name="light_blue_400">#FF29B6F6</color>
```
Comment (lechld) [inline app/src/main/res/values/colors.xml:64]: please remove colours

File: app/src/main/res/values/dimens.xml [hunk 13]
```diff
@@ -6,6 +6,7 @@
     <dimen name="default_margin">16dp</dimen>
     <!-- Half margin of default_margin. Can be helpful if we items next to each other. Eg: RecyclerView. -->
     <dimen name="default_margin_half">8dp</dimen>
+    <dimen name="text_margin">16dp</dimen>
```
Comment (lechld) [inline app/src/main/res/values/dimens.xml:9]: I haven't seen where you use that. But use `default_margin` instead and remove please

File: app/src/main/res/values/styles.xml [hunk 14]
```diff
@@ -0,0 +1,6 @@
+<resources>
```
Comment (lechld) [inline app/src/main/res/values/styles.xml:1]: please remove

File: build.gradle [hunk 15]
```diff
@@ -1,4 +1,11 @@
 buildscript {
+    ext {
+        kotlin_version = '1.8.20'
+    }
+    dependencies {
+        classpath "androidx.navigation:navigation-safe-args-gradle-plugin:2.5.3"
```
Comment (lechld) [inline build.gradle:6]: we don't use navigation library currently. So we don't need safe args. I'd suggest to revert changes in this file, it should work without them.

**Review by lechld — APPROVED** [review]
Good work, just some stuff to cleanup (guess leftovers from experimenting with drag&drop)

**Comment by lechld** [comment]
@ct222ct please create follow up issue to add missing tests

**Comment by Kirschzelle** [comment]
So, i went in and fixed sonar not building. Just straight up errors in the IDE.

`-->` If you rename, use refactor, else you get stuff like this. (Check last commit from me, basically you renamed from Discover to Mainscreen, but didnt change depndencies etc. etc. --> You use refactor for that, so it changes everything for you.) 
Please care for errors your IDE shows you. You must not commit with yellow warnings, these are red. Stuff like this should not get past you.
Screenshot_2023-04-24_at_18 13 15
(also not using junit import that does not resolve + unused imports --> auto format + auto resolve imports on keybind for that)

`-->` 2 Tests are failing. Please actually execute them with jacoco. (If you don't know where to run jacoco write me and I will show you.) Either delete them for now or rewrite them. Good job on getting on to write the tests, but at the same time compiler warnings should really be taken seriously. If you didn't see the warnings in the IDE you should have noted Sonar not building. (what got me to go take a look)

Ps. Please try using refactor feature, you will definitely need it a lot! `-->` Rename it to discover again, since that was the correct name.

You are free to merge as long as sonar cloud finished without errors and you renamed it.

File: .idea/androidTestResultsUserPreferences.xml (hunk 1 above)
Comment (ct222ct) [reply .idea/androidTestResultsUserPreferences.xml]: File deleted

File: .idea/deploymentTargetDropDown.xml (hunk 2 above)
Comment (ct222ct) [reply .idea/deploymentTargetDropDown.xml]: File deleted

File: app/src/main/res/values/attrs_my_view.xml (hunk 11 above)
Comment (ct222ct) [reply app/src/main/res/values/attrs_my_view.xml]: Removed Test Code

File: app/src/main/res/values/dimens.xml (hunk 13 above)
Comment (ct222ct) [reply app/src/main/res/values/dimens.xml]: deleted line 9

File: app/src/main/res/values/styles.xml (hunk 14 above)
Comment (ct222ct) [reply app/src/main/res/values/styles.xml]: removed

File: build.gradle (hunk 15 above)
Comment (ct222ct) [reply build.gradle]: removed line 6

**Comment by ct222ct** [comment]
Merge pull request (Mainscreen) to Master

**Pull request merged**